examples/siro_sandbox/gui_pick_helper.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import magnum as mn
+import numpy as np
+
+
+class GuiPickHelper:
+    def __init__(self, gui_service, agent_idx):
+        self._sandbox_service = gui_service
+        self._agent_idx = agent_idx
+        self._rom = self._get_sim().get_rigid_object_manager()
+        self.obj_ids = self._get_sim()._scene_obj_ids
+
+    def _get_sim(self):
+        return self._sandbox_service.sim
+
+    def _closest_point_and_dist(self, origin, direction_vector, points):
```

Comment:
I stared at this a while and couldn't make sense of what you're doing. Can you add a comment? Something like finding the point nearest to the plane formed by the origin and the direction vector?